src/parser.rs

```diff
@@ -621,6 +629,18 @@ mod tests {
         }
     }
 
+    #[test]
+    fn bodystructure() {
+        const RESPONSE: &[u8] = b"* 15 FETCH (BODYSTRUCTURE (\"TEXT\" \"PLAIN\" (\"CHARSET\" \"iso-8859-1\") NIL NIL \"QUOTED-PRINTABLE\" 1315 42 NIL NIL NIL NIL))\r\n";
+        match parse_response(RESPONSE) {
+            IResult::Done(_, Response::Fetch(_, attrs)) => {
+                let body = &attrs[0];
+                assert!(if let &AttributeValue::Bodystructure { .. } = body { true } else { false }, "body = {:?}", body);
```

Comment:
This isn't very easy to grok. It's a fairly long expression with a sizable amount of complexity, all on a single line. It would probably be better to just assign the boolean to a local variable, then `assert!()` that.